### PR TITLE
fs: Wipe all signatures when running wipe with wipe_all set to TRUE

### PR DESCRIPTION
**Pull request by vojtechtrefny**
When wiping filesystem signatures from a device, we are not trying
to wipe all signatures, but only signatures matching the filesystem
name -- so for example all "ntfs" signatures.
This makes sense for FAT with multiple "vfat" signatures, but
NTFS has one "ntfs" signature and one nested "dos" partition table
which we also need to remove.
This change makes sure all signatures (no only filesystem ones and
not only these matching the filesystem name) are wiped from the
device.

-----

I'm working on some filesystem plugin test changes and I noticed there are some missing tests for NTFS so I added those and found this issue. I'm posting this for a separate review because it's quite big change and it can potentially break something. Alternative approach would be to change this behaviour only for NTFS (the `wipe_fs` function is internal so we can add an extra argument for this specific use case).

### CODE REVIEW

**Review by tbzatek — APPROVED** [review]
Good catch.

What about issuing BLKDISCARD as well? Most filesystems do that automatically, a little help might not harm perhaps.

File: src/plugins/fs/common.c [hunk 1]
```diff
@@ -192,7 +192,7 @@ wipe_fs (const gchar *device, const gchar *fs_type, gboolean wipe_all, GError **
     blkid_reset_probe (probe);
 
     if (wipe_all) {
-        has_fs_type = has_fs (probe, device, fs_type, error);
+        has_fs_type = has_fs (probe, device, NULL, error);
```
Comment (vpodzime) [inline src/plugins/fs/common.c:195]: The name of the variable no longer makes sense. And maybe the whole code can be simplified?

**Review by vpodzime — CHANGES_REQUESTED** [review]

File: src/plugins/fs/common.c (hunk 1 above)
Comment (vojtechtrefny) [reply src/plugins/fs/common.c]: Good point, fixed.

**Review by vpodzime — APPROVED** [review]
Looks good to me now. Just please check that the documentation is still valid for the new behavior.

**Comment by vojtechtrefny** [comment]
> Looks good to me now. Just please check that the documentation is still valid for the new behavior.

This is an internal documentation so there wasn't a docstring but I added one with the `wipe_all` explanation for future reference.

**Pull request merged**